fe/fe-core/src/test/java/com/starrocks/server/WarehouseManagerTest.java

```diff
@@ -258,5 +267,29 @@ public Warehouse getWarehouse(long warehouseId) {
         } catch (ErrorReportException e) {
             Assert.assertEquals(1, 2);   // can not be here
         }
+
+        new MockUp<RunMode>() {
+            @Mock
+            public RunMode getCurrentRunMode() {
+                return RunMode.SHARED_DATA;
+            }
+        };
+        try {
+            OlapScanNode scanNode = newOlapScanNode();
+            Partition partition = new Partition(123, "aaa", null, null);
+            MaterializedIndex index = new MaterializedIndex(1, MaterializedIndex.IndexState.NORMAL);
+            scanNode.addScanRangeLocations(partition, partition, index, Collections.emptyList(), 1);
+            Assert.fail();
+        } catch (ErrorReportException e) {
+            Assert.assertEquals("No alive backend or compute node in warehouse null.", e.getMessage());   // can not be here
```

Comment:
it is excepted to be reaching here, right? so the comment is incorrect.

And also you can use Assert.assertThrows() to assert a statement must be thrown exception.

```
ErrorReportException ex = Assert.assertThrows(ErrorReportException.class, () -> scanNode.addScanRangeLocations(partition, partition, index, Collections.emptyList(), 1));
// check exception info
...
```

Reply:
Modified.